From 92935527f602d8450696b4176f72e0bb9b7813a5 Mon Sep 17 00:00:00 2001
From: Ecco park <eccopark@broadcom.com>
Date: Fri, 3 Oct 2014 13:13:43 -0700
Subject: [PATCH 08/27] net: wireless: bcmdhd: reduced the MAX_CNTL_TX_TIMEOUT
 to 1

Currently the recovery mechanism is not triggered because Wi-Fi
Framework will turn on Wi-Fi again as soon as we get resumed on
timeout for the first time.
Currently the recovery mechanism is triggered if we get resumed
on timeout two times in sequence.
In order to trigger recovery mechanism, we need to the count from 2 to 1.

b/17755061 Wi-Fi stuck in Off position

Change-Id: I86a950243c742e561c3ac8a8722bd9a760331410
Signed-off-by: Ecco park <eccopark@broadcom.com>
---
 drivers/net/wireless/bcmdhd/Makefile   | 1 +
 drivers/net/wireless/bcmdhd/dhd_pcie.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index e24df53..27bb4a8 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -159,6 +159,7 @@ ifneq ($(CONFIG_BCMDHD_PCIE),)
   DHDCFLAGS += -DCUSTOM_DHD_WATCHDOG_MS=0
 
   DHDCFLAGS += -DSAR_SUPPORT
+  DHDCFLAGS += -DMAX_CNTL_TX_TIMEOUT=1
 ifneq ($(CONFIG_ARCH_MSM),)
   DHDCFLAGS += -DMSM_PCIE_LINKDOWN_RECOVERY
 endif
diff --git a/drivers/net/wireless/bcmdhd/dhd_pcie.c b/drivers/net/wireless/bcmdhd/dhd_pcie.c
index c8bcbc4..a4cfdbc 100644
--- a/drivers/net/wireless/bcmdhd/dhd_pcie.c
+++ b/drivers/net/wireless/bcmdhd/dhd_pcie.c
@@ -1185,6 +1185,7 @@ int dhd_bus_rxctl(struct dhd_bus *bus, uchar *msg, uint msglen)
 	if (bus->dhd->rxcnt_timeout >= MAX_CNTL_TX_TIMEOUT) {
 #ifdef MSM_PCIE_LINKDOWN_RECOVERY
 		bus->islinkdown = TRUE;
+		DHD_ERROR(("PCIe link down\n"));
 #endif /* SUPPORT_LINKDOWN_RECOVERY */
 		return -ETIMEDOUT;
 	}
-- 
1.9.1

